dobi.yaml

```diff
@@ -0,0 +1,93 @@
+
+meta:
```

Comment:
I see what you're saying. I think the difference is that I would never run the fabfile in the first place.

If I'm new to a project and I just want to try it out,  I don't want to install a bunch of things on my host that I may never use again, or that I might not use for a while.  I don't want to have to install updates for them, and worry about them running services. And I don't want to type my root password for it, without examining the file first (which is huge! so would take a long time). And that's assuming I'm running `ubuntu` and the `fabfile` "might" work.

So instead I want to install things in Docker. That way I can just `docker rmi` when I'm done. Even if I don't remove them, I can easily contribute things to the project without worrying about breaking it for OSX, because everyone would be running the same image. If I install on my host, I have to guess at how to make it work for OSX, redhat, etc.

>  I want to build the docker image myself 

Right, I think I would always want to build it myself, because I want to be able to test out my changes.

So I could `docker build`, and the `Dockerfile` could just run `fab ...` except that is very slow. Any change to the `fabfile` would invalidate the entire cache, which would incur another 20-30min rebuild of the image.

Using `dobi` (and having separate steps for things outside of the `fabfile`) means that most of the time the `docker build` cache will be used, and I'll be able to  pickup changes in just a few seconds, instead of 20-30min.

(`fabric` also requires an ssh server. Running an SSH server in a container is kind of an anti-pattern, but it's still possible, and there are hacks around it as well, which I think I had in my earlier PR).